frontend/dockerfile/dockerfile_test.go

```diff
@@ -6764,78 +6767,138 @@ ARG BUILDKIT_SBOM_SCAN_STAGE=true
+			Detail: "Stage names should be lowercase",
+			Level:  1,
+		},
+		{
+			Short:  "Lint Rule 'FromAsCasing': 'as' and 'FROM' keywords' casing do not match (line 6)",
+			Detail: "The 'as' keyword should match the case of the 'from' keyword",
+			Level:  1,
+		},
+	})
+
+	dockerfile = []byte(`
+# warning: 'AS' should match 'from' cmd casing.
+from scratch AS base
+
+from scratch as base2
+`)
+	checkLintWarnings(t, sb, dockerfile, []expectedLintWarning{
+		{
+			Short:  "Lint Rule 'FromAsCasing': 'AS' and 'from' keywords' casing do not match (line 3)",
+			Detail: "The 'as' keyword should match the case of the 'from' keyword",
+			Level:  1,
+		},
+	})
+}
 
-	// empty line in here is intentional to cause line continuation warning
+func testLintNoEmptyContinuations(t *testing.T, sb integration.Sandbox) {
 	dockerfile := []byte(`
 FROM scratch
+# warning: empty continuation line
 COPY Dockerfile \
 
+.
+COPY Dockerfile \
 .
 `)
 
-	dir := integration.Tmpdir(
-		t,
-		fstest.CreateFile("Dockerfile", dockerfile, 0600),
-	)
-
-	c, err := client.New(sb.Context(), sb.Address())
-	require.NoError(t, err)
-	defer c.Close()
+	checkLintWarnings(t, sb, dockerfile, []expectedLintWarning{
+		{
+			Short:  "Lint Rule 'NoEmptyContinuations': Empty continuation line (line 6)",
+			Detail: "Empty continuation lines will become errors in a future release",
+			URL:    "https://github.com/moby/moby/pull/33719",
+			Level:  1,
+		},
+	})
+}
 
-	status := make(chan *client.SolveStatus)
-	statusDone := make(chan struct{})
-	done := make(chan struct{})
+func testSelfConsistentCommandCasing(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+# warning: 'FROM' should be either lowercased or uppercased
+From scratch as base
+FROM scratch AS base2
+`)
+	checkLintWarnings(t, sb, dockerfile, []expectedLintWarning{
+		{
+			Short:  "Lint Rule 'SelfConsistentCommandCasing': Command 'From' should be consistently cased (line 3)",
+			Detail: "Commands should be in consistent casing (all lower or all upper)",
+			Level:  1,
+		},
+	})
+	dockerfile = []byte(`
+# warning: 'FROM' should be either lowercased or uppercased
+frOM scratch as base
+from scratch as base2
+`)
+	checkLintWarnings(t, sb, dockerfile, []expectedLintWarning{
+		{
+			Short:  "Lint Rule 'SelfConsistentCommandCasing': Command 'frOM' should be consistently cased (line 3)",
+			Detail: "Commands should be in consistent casing (all lower or all upper)",
+			Level:  1,
+		},
+	})
+}
 
-	var warnings []*client.VertexWarning
+func testFileConsistentCommandCasing(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+FROM scratch
 
-	go func() {
-		defer close(statusDone)
-		for {
-			select {
-			case st, ok := <-status:
-				if !ok {
-					return
-				}
-				warnings = append(warnings, st.Warnings...)
-			case <-done:
-				return
-			}
-		}
-	}()
+# warning: 'copy' should match first command's casing (in this case, 'FROM's casing)
+copy Dockerfile /foo
```

Comment:
Add another lowercase command as well to test that this warning only appears once per file and if I have `FROM` followed by 20 `copy/run` then I don't get 20 warnings.

Reply:
I don't think that would be correct behavior. It's maybe noisy if there's a `FROM` followed by 20 `copy/run`, but having it happen only once per file means that if I have, say, two instances where I have erroneously used alternate casing, then I could lint it, fix EVERY lint warning it reports, and then STILL fail the subsequent lint, but now with different warnings.

If I had, say, 3 inconsistent commands out of 100, it would be maddening to have to run the linter 3 different times in order to be able to have all of these reported.

I'm going to opt instead to be more sophisticated in how we check for consistent command casing throughout the file to compare whether a command matches the most used casing, which will prevent this scenario!